### PR TITLE
Callcenter user controller

### PR DESCRIPTION
**Pull request by daveRanjan**

### CODE REVIEW

File: src/main/java/com/clogic/resourceserver/service/impl/callcenter/CallCenterServiceImpl.java [hunk 1]
```diff
@@ -52,20 +52,20 @@ public CallCenterResponseDto createCallCenter(CallCenterRequestDto callCenterReq
...
-        userService.saveUser(user);
+        userService.saveUser(savedClient.getId(), user);
 
         callCenterResponseDto.setUser(UserHelper.userToDto(user));
         callCenterResponseDto.setClient(ClientResponseDto.clientToDto(clientService.getClientFromClientId(client.getClientId())));
 
         ClientUser clientUser = new ClientUser();
         clientUser.setClientId(clientService.getClientFromClientId(client.getClientId()).getId());
-        clientUser.setUserId(userService.getUserByEmailId(user.getEmailId()).getUserId());
+        clientUser.setUserId(userService.getUserByEmailId(null, user.getEmailId()).getUserId());
```
Comment (nishantt95) [inline src/main/java/com/clogic/resourceserver/service/impl/callcenter/CallCenterServiceImpl.java]: why null?

File: src/main/java/com/clogic/resourceserver/service/impl/user/UserServiceImpl.java [hunk 2]
```diff
@@ -1,36 +1,120 @@
...
+    private String salt;
+
+    @Value("${encryption.users.iterations}")
+    private Integer iterations;
+
+    public List<User> getUsers(Long centerId) {
+        List<User> users;
+        if (centerId != null) {
+            users = clientService.getClientFromId(centerId.intValue()).getUsers().stream().collect(Collectors.toList());
+        } else {
```
Comment (nishantt95) [inline src/main/java/com/clogic/resourceserver/service/impl/user/UserServiceImpl.java]: `clientService.getClientFromId(centerId.intValue()).getUsers().stream().collect(Collectors.toList());`

??

File: src/main/java/com/clogic/resourceserver/service/impl/user/UserServiceImpl.java [hunk 3]
```diff
@@ -1,36 +1,120 @@
...
-    public void saveUser(User user){
-        userRepository.save(user);
+    public User saveUser(Integer centerId, User user) {
+
+        String generatedPassword = StringUtils.isNotBlank(user.getPassword()) ? user.getPassword() : RandomStringUtils.randomAlphabetic(10);
+
+        user.setPassword(SecurityUtils.generateHash(generatedPassword, salt, iterations));
+        user.setSalt(salt);
+        user.setIterations(iterations);
+        user.setContactCenter(centerId);
```
Comment (nishantt95) [inline src/main/java/com/clogic/resourceserver/service/impl/user/UserServiceImpl.java]: **Contact Center** in user is equal to **call center Id** ?

File: src/main/java/com/clogic/resourceserver/service/impl/user/UserServiceImpl.java (hunk 2 above)
Comment (daveRanjan) [reply src/main/java/com/clogic/resourceserver/service/impl/user/UserServiceImpl.java]: Java 8 syntax for converting Set to List

**Pull request merged**